typing/datarepr.ml

```diff
@@ -315,3 +317,29 @@ let unboxed_labels_of_type ty_path decl =
         labels rep decl.type_private
   | Type_record _
   | Type_variant _ | Type_abstract _ | Type_open -> []
+
+let constructor_unbound_type_vars_excluding_row_variables cstr =
```

Comment:
Much of this computation duplicates work in `for_boxed_variant`. Maybe just inline this? That is, you traverse the result args to find all the bare Tvars, stored in a list. This list will be matched with the params in the call to `apply`. But it also can be turned into a set and subtracted from the free variables of the entire constructor. This approach has the nice property of making it even clearer that every variable free in a constructor's fields gets substituted away by the `apply`, either to a param or to a `Tof_kind`.

Reply:
I think you'd probably need to pass `Ctype.free_vars` into `for_boxed_variant`. That's a small shame.... but certainly no worse than the forward-declaration of this function.